packages/core/src/builtin-tasks/moleculec.ts

```diff
@@ -0,0 +1,278 @@
+  }
+}
+
+function checkMoleculecExist() {
+  try {
+    execSync('moleculec -help')
+  } catch {
+    console.warn('Warning: moleculec not found, start install moleculec')
+    installMoleculec()
+  }
+}
+
+task('moleculec', 'Create molecule interface with schema file')
+  .addParam('schema-file', 'Provide a schema file to compile.', '', paramTypes.string)
+  .addParam(
+    'language',
+    'Specify a language, then generate source code for the specified language and output the generated code',
+    '',
+    paramTypes.string,
+  )
+  .addParam(
+    'output',
+    "The generated code will output path, Default output to the schema-file's directory",
+    '',
+    paramTypes.string,
+  )
+  .addParam('install', 'install moleculec with cargo', false, paramTypes.boolean)
+  .setAction(async ({ language, schemaFile, output }: Args) => {
+    if (!language || !schemaFile) {
+      console.error('Error: The following required arguments were not provided:')
+      if (!language) console.error('--language <language>')
+      if (!schemaFile) console.error('--schema-file <schema-file>')
+      return
+    }
+    if (!fs.existsSync(schemaFile) || !fs.statSync(schemaFile).isFile()) {
+      console.error(`Error: schema-file [${schemaFile}] should be a file`)
+      return
+    }
+    if (language !== 'ts' && language !== 'rust' && language !== 'c') {
+      console.error(`Error: language [${language}] should be one of ts/rust/c`)
+      return
+    }
+    await checkMoleculecExist()
+    console.info('start generate...')
+    const outputPath =
+      output ||
+      path.format({
+        dir: path.dirname(schemaFile),
+        name: path.basename(schemaFile, '.mol'),
+        ext: extMap[language],
+      })
+    if (language !== 'ts') {
+      try {
+        execSync(`moleculec --schema-file ${schemaFile} --language ${language} > ${outputPath}`)
+        console.info('generate success...')
+      } catch (error) {
+        console.error(error)
+      }
+      return
+    }
+    try {
+      const schemaStructJSON = execSync(`moleculec --schema-file ${schemaFile} --language - --format json`).toString()
+      const { declarations } = JSON.parse(schemaStructJSON) as Schema
+      const declarationsMaps: Record<string, Declaration> = {}
+      declarations.forEach((v) => {
+        declarationsMaps[v.name] = v
+      })
+      const outputStream = fs.createWriteStream(outputPath)
+      declarations.map((v) => {
+        outputStream.write(
+          `export type ${v.name}Type = ${JSON.stringify(generateSchemaToInterface(v, declarationsMaps), undefined, 2)}`,
+        )
+        outputStream.write('\r\n')
+        outputStream.write(
+          `export const ${v.name}: ${v.name}Type = ${JSON.stringify(
+            generateSchemaToInterface(v, declarationsMaps),
+            undefined,
+            2,
+          )}`,
+        )
+        outputStream.write('\r\n\r\n')
+      })
+      outputStream.close()
+      console.info('generate success...')
```

Comment:
it is better to extract the section as an independent function to make it testable

```ts

function codegenTs(mol: string): string {
  // ...
}

test('codegen to TypeScript', () => {
  test(codegenTs(mol)).toMatchSnapshot();
})

```

Reply:
I have extracted the section, and I will add test cases for these functions.